### PR TITLE
Add parent account fields to analytics_accounts.

### PR DESCRIPTION
**Pull request by beccagaspard**
I added the parent account id, parent account name, and parent account external key to the analytics_accounts table so that it can be used in reporting.  Please let me know if any changes are needed.  Thanks! 😀

### CODE REVIEW

**Comment by sbrossie** [comment]
That seems fairly straightforward, will let @pierre to review as well before me merge. 

@beccagaspard Did you have a chance to start the plugin and test that it still works correctly both for accounts that have a NULL/non NULL `parent_account_id` ?

**Comment by beccagaspard** [comment]
@sbrossie Yup, I ran the plugin and tested both cases (parent and no parent) and everything seemed to be working as expected with the correct data populated in `analytics_accounts` 👍

File: src/main/java/org/killbill/billing/plugin/analytics/dao/model/BusinessAccountModelDao.java [hunk 1]
```diff
@@ -498,6 +527,15 @@ public boolean equals(final Object o) {
         if (oldestUnpaidInvoiceId != null ? !oldestUnpaidInvoiceId.equals(that.oldestUnpaidInvoiceId) : that.oldestUnpaidInvoiceId != null) {
             return false;
         }
+        if (parentAccountExternalKey != null ? !parentAccountExternalKey.equals(that.parentAccountExternalKey) : that.stateOrProvince != null) {
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/analytics/dao/model/BusinessAccountModelDao.java]: Typo (`stateOrProvince`).

File: src/main/resources/migration/V20170208000000__add_parent_account.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+alter table analytics_accounts add parent_account_id varchar(36) default null after account_external_key;
+alter table analytics_accounts add parent_account_name varchar(100) default null after parent_account_id;
+alter table analytics_accounts add parent_account_external_key varchar(50) default null after parent_account_name;
```
Comment (pierre) [inline src/main/resources/migration/V20170208000000__add_parent_account.sql]: I know we still use 50 elsewhere (e.g. `account_external_key`), but better to make it 255 instead so it matches the main DDL (we should also probably migrate other columns at some point...).

See https://github.com/killbill/killbill/issues/604.

File: src/main/resources/org/killbill/billing/plugin/analytics/ddl.sql [hunk 3]
```diff
@@ -155,6 +155,9 @@ create table analytics_accounts (
 , account_id varchar(36) default null
 , account_name varchar(100) default null
 , account_external_key varchar(50) default null
+, parent_account_id varchar(36) default null
+, parent_account_name varchar(100) default null
+, parent_account_external_key varchar(50) default null
```
Comment (pierre) [inline src/main/resources/org/killbill/billing/plugin/analytics/ddl.sql]: Same remark.

**Comment by pierre** [comment]
Thanks! 🎉

File: src/main/resources/migration/V20170208000000__add_parent_account.sql (hunk 2 above)
Comment (pierre) [inline src/main/resources/migration/V20170208000000__add_parent_account.sql]: Done in https://github.com/killbill/killbill-analytics-plugin/commit/f097ca4458b07e05e94bf2f2cb09a4e9878ab137.

**Pull request merged**